src/tests/EShopworld.WorkerProcess.IntegrationTests/DistributedLockTests.cs

```diff
@@ -0,0 +1,114 @@
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace EShopworld.WorkerProcess.IntegrationTests
+{
+    public class DistributedLockTests
+    {
+        private readonly ServiceProvider _serviceProvider;
+        private readonly ITestOutputHelper _output;
+
+        public DistributedLockTests(ITestOutputHelper output)
+        {
+            var configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json", false, true)
+                .Build();
+
+            var serviceCollection = new ServiceCollection();
+            ConfigureServices(serviceCollection, configuration);
+
+            _serviceProvider = serviceCollection.BuildServiceProvider();
+
+            _output = output;
+        }
+
+        [Fact, IsIntegration]
+        public void AcquireLock_ForNewLock_ShouldNotThrowException()
+        {
+            // Arrange
+            Func<Task> act = async () =>
+            {
+                var distributedLock = _serviceProvider.GetService<IDistributedLock>();
+                using (await distributedLock.Acquire("test1"))
+                {
+                    // create a recurring job
+                }
+            };
+
+            // Act - Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact, IsIntegration]
+        public void AcquireLock_ForExistingLock_ShouldThrowException()
+        {
+            // Arrange
+            Func<Task> act = async () =>
+            {
+                var distributedLock = _serviceProvider.GetService<IDistributedLock>();
+                using (await distributedLock.Acquire("test2"))
+                {
+                    using (await distributedLock.Acquire("test2"))
+                    {
+                        // create a recurring job
+                    }
+                }
+            };
+
+            // Act - Assert
+            act.Should().Throw<DistributedLockNotAcquiredException>();
+        }
+
+        private void ConfigureServices(IServiceCollection services, IConfigurationRoot configuration)
+        {
+            services.AddOptions();
+            services.Configure<CosmosDbConnectionOptions>(configuration.GetSection("CosmosDbConnectionOptions"));
+            services.Configure<CosmosDataStoreOptions>(configuration.GetSection("CosmosDataStoreOptions"));
+            services.Configure<TelemetrySettings>(configuration.GetSection("Telemetry"));
+
+
+            var serviceProvider = services.BuildServiceProvider();
+
+            var telemetry = serviceProvider
+                .GetService<IOptions<TelemetrySettings>>();
+
+#pragma warning disable 618
+            var bb = new BigBrother(telemetry.Value.InstrumentationKey, telemetry.Value.InternalKey);
```

Comment:
Can use BigBrother.CreateDefault if you wanna get rid of the warning

Reply:
Done.